### PR TITLE
Add support for feature to vary and permitted range in counterfactual manager

### PR DESCRIPTION
**Pull request by gaugup**

### CODE REVIEW

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 1]
```diff
@@ -145,7 +152,7 @@ def add(self,
             desired_class=None,
             desired_range=None,
             permitted_range=None,
-            features_to_vary=None,
+            features_to_vary='all',
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:155]: Is `all` equivalent to `None` in diceml?

File: responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                assert np.all(
+                    cf_obj.cf_examples_list[0].final_cfs_df[feature_name] ==
+                    X_test.iloc[0:1][feature_name][0])
+            else:
+                assert np.any(
+                    cf_obj.cf_examples_list[0].final_cfs_df[feature_name] !=
+                    X_test.iloc[0:1][feature_name][0])
+
+    @pytest.mark.parametrize('feature_importance', [True, False])
+    def test_counterfactual_permitted_Range(self, feature_importance):
```
Comment (gregorybchris) [inline responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py:61]: Nit: lowercase `range`

**Review by gregorybchris — APPROVED** [review]

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:155]: No I got it wrong in first place. The default in dice-ml was 'all''.

https://github.com/interpretml/DiCE/blob/004dbd508ea2b801b5f3ca26a87a25eafc19e1f0/dice_ml/explainer_interfaces/explainer_base.py#L50

File: responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py:61]: Done

**Pull request merged**